ui-modules/utils/icon-generator/icon-generator.js

```diff
@@ -104,21 +117,49 @@ function IconGenerator(cache, background, margin, size) {
 
 export function iconServiceProvider() {
     return {
-        $get: ['$q', '$http', 'iconGenerator', function ($q, $http, iconGenerator) {
-            return new IconService($q, $http, iconGenerator, new SessionsStorageWrapper(CACHE_NAME));
+        $get: ['$q', '$http', 'iconGenerator', '$log', function ($q, $http, iconGenerator, $log) {
+            return new IconService($q, $http, iconGenerator, $log, new SessionsStorageWrapper(CACHE_NAME));
         }]
     }
 }
 
-function IconService($q, $http, iconGenerator, cache) {
+function IconService($q, $http, iconGenerator, $log, cache) {
     this.get = getIcon;
-    function getIcon(id) {
+    function getIcon(entityOrTypeId, doNotAutogenerate) {
```

Comment:
As far as I can see, the there is no called using the `doNotAutogenerate`. It seems to be used only on the pipe.

Reply:
it's used in `entity-tree.directive.js`:

```
iconService.get($scope.entity, true).then(value => $scope.iconUrl = value);
```